packages/custom_lint_builder/lib/src/custom_analyzer_converter.dart

```diff
@@ -72,23 +72,28 @@ class CustomAnalyzerConverter {
   /// start column. If an analysis [options] is provided then the severities of
   /// the errors will be altered based on those options.
   List<plugin.AnalysisError> convertAnalysisErrors(
-      List<analyzer.AnalysisError> errors,
-      {analyzer.LineInfo? lineInfo,
-      analyzer.AnalysisOptions? options}) {
-    var serverErrors = <plugin.AnalysisError>[];
-    for (var error in errors) {
-      var processor = analyzer.ErrorProcessor.getProcessor(options, error);
-      if (processor != null) {
-        var severity = processor.severity;
-        // Errors with null severity are filtered out.
-        if (severity != null) {
-          // Specified severities override.
-          serverErrors.add(convertAnalysisError(error,
-              lineInfo: lineInfo, severity: severity));
-        }
-      } else {
-        serverErrors.add(convertAnalysisError(error, lineInfo: lineInfo));
+    List<analyzer.AnalysisError> errors, {
```

Comment:
This feels like an odd place to have this logic.

But I can't immediately think of a better location due to AnalysisError being immutable.

Reply:
I'd rather not change this file, as it is forked from analyzer. I want to keep it close to the source, for easy upgrade.


Could we maybe add a separate util such that:
https://github.com/invertase/dart_custom_lint/blob/f44a7cb2c20f26d2c960fa77736b7e86ebd12a16/packages/custom_lint_builder/lib/src/client.dart#L994-L995

Does something like:

```dart
...convertAnalysisError(...).map((error) => error.copyWith(error: <apply error config>))
```

Then the file stays untouched, and the location of the logic is in a more sensible place.